core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnInsertExec.scala

```diff
@@ -329,6 +347,16 @@ case class ColumnInsertExec(_child: SparkPlan, partitionColumns: Seq[String],
        |${if (numInsertedRowsMetric eq null) ""
           else s"$numInsertedRowsMetric.${metricAdd(numInsertions)};"}
        |${consume(ctx, Seq(ExprCode("", "false", numInsertions)))}
+       |success = true;
+       |}
+       |finally {
+       |if (txStarted) {
+       |  if(success)
+       |    com.pivotal.gemfirexd.internal.engine.Misc.getGemFireCache().getCacheTransactionManager().commit();
```

Comment:
The same way we started a transaction based on embedded or connector mode, we should rollback commit also in same fashion. The same code will be used for both mode and accessing a cache here will fail.